### DISCUSSION TITLE
Push notifications: Receive the firebase FCM token from iOS instead of the raw APNS token

### DISCUSSION THREAD

**Discussion by ptmkenny** [ionic-team/capacitor#4672] [Q&A] [answered]
I'm trying to set up notifications in iOS and I want to send the firebase FCM token to my server instead of the APNS token. 

[The documentation states](https://capacitorjs.com/docs/guides/push-notifications-firebase):

`If you would like to recieve the firebase FCM token from iOS instead of the raw APNS token, you will need to also change your AppDelegate.didRegisterForRemoteNotificationsWithDeviceToken code to look like this:`

Where do I put this code?  There is no `AppDelegate.didRegisterForRemoteNotificationsWithDeviceToken` in `AppDelegate.swift`, and when I search for the string `AppDelegate.didRegisterForRemoteNotificationsWithDeviceToken`, it is referenced in many places...

**Comment by kensodemann** [comment, marked as answer]
Likely referring to the one added if you follow the directions in the plugin docs: https://capacitorjs.com/docs/apis/push-notifications#ios

The doc that you reference needs to be cleaned up a bit from what I see.

**Reply by jcesarmobile** [reply]
Yeah, that code used to be on the app template, but was removed on Capacitor 3 and documented in push plugin to add it only if you need it, so the guide should be updated.
